awx/main/tasks.py

```diff
@@ -396,47 +406,74 @@ def purge_old_stdout_files():
             logger.debug("Removing {}".format(os.path.join(settings.JOBOUTPUT_ROOT, f)))
 
 
+@task(queue=get_local_queuename)
+def check_heartbeat(node):
+    AWXReceptorJob.check_heartbeat(node)
+
+
+def discover_receptor_nodes():
+    ctl = ReceptorControl(RECEPTOR_SOCK)
+    connections = ctl.simple_command('status')['Advertisements']
+    nowtime = now()
+    for ad in connections:
+        hostname = ad['NodeID']
+        commands = ad['WorkCommands'] or []
+        if 'ansible-runner' not in commands:
```

Comment:
This discovery of advertisements that container an `ansible-runner` command is just an informal convention.  We could change the way discovery works, but as a proof of concept for delineating receptor nodes capable of Ansible execution, this approach made sense to me.